lighthouse-core/runner.js

```diff
@@ -230,8 +230,13 @@ class Runner {
       requestedUrl,
       settings: runnerOpts.config.settings,
     };
-    const artifacts = await GatherRunner.run(runnerOpts.config.passes, gatherOpts);
-    return artifacts;
+    const gatherPromise = GatherRunner.run(runnerOpts.config.passes, gatherOpts);
+    await Promise.race([
+      gatherPromise,
+      GatherRunner.getGatherTerminatedPromise(driver),
```

Comment:
Congratulations this is one of the only forked areas where we need to worry about duplication for FR 🎉  lucky you! 😃 

would you be able to add this to the snapshot runner and use the shared `FRTransitionalDriver` type instead? :)

living in gather-runner feels a little weird at that point, but I don't have a clean new place or suggestion for it. ideas?